clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -627,8 +623,7 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 "Branch target not immediately after branch");
         }
         // Invariant: non-explicit control flow (i.e., kernel-mediated) is indicated
-        // by markers.  Checks are relaxed if a kernel event occurred so prev_instr_
-        // is sufficient for now.
+        // by markers.
```

Comment:
But it does seem worth explaining why prev_instr_ is used here instead of the context one: because after a signal the interruption and resumption checks are done elsewhere and this call turns into a nop essentially (right?).